cmd/controller-manager/app/helper_test.go

```diff
@@ -60,6 +60,20 @@ func TestIsControllerEnabled(t *testing.T) {
 			disabledByDefaultControllers: []string{"delta", "echo"},
 			expected:                     false,
 		},
+		{
+			name:                         "on by all except",
+			controllerName:               "alpha",
+			controllers:                  []string{"*", "-charlie"},
+			disabledByDefaultControllers: []string{"delta", "echo"},
+			expected:                     true,
+		},
+		{
+			name:                         "off by all except",
+			controllerName:               "charlie",
+			controllers:                  []string{"*", "-charlie"},
+			disabledByDefaultControllers: []string{"delta", "echo"},
```

Comment:
does it diff with the case[off by name]?

Reply:
Yes, all except {"*", "-charlie"} is different from {"alpha", "bravo" "-charlie"}, both in semantical and logical. The var `hasStar` in the function is different, there are `true` and `false` respectively

Reply:
the key is `-Charlie` in controllers